ndb/memcache_client.py

```diff
@@ -0,0 +1,220 @@
+    if not isinstance(time, (int, long)):
+      raise TypeError('time must be a number; received %r' % time)
+    if namespace is None:
+      namespace = namespace_manager.get_namespace()
+    options = ('set', time, namespace, deadline)
+    batcher = self.memcache_set_batcher
+    if use_cache:
+      return batcher.add_once((key, value), options)
+    else:
+      return batcher.add((key, value), options)
+
+  def memcache_add(self, key, value, time=0, namespace=None, deadline=None):
+    if not isinstance(key, basestring):
+      raise TypeError('key must be a string; received %r' % key)
+    if not isinstance(time, (int, long)):
+      raise TypeError('time must be a number; received %r' % time)
+    if namespace is None:
+      namespace = namespace_manager.get_namespace()
+    return self.memcache_set_batcher.add((key, value),
+                                          ('add', time, namespace, deadline))
+
+  def memcache_replace(self, key, value, time=0, namespace=None, deadline=None):
+    if not isinstance(key, basestring):
+      raise TypeError('key must be a string; received %r' % key)
+    if not isinstance(time, (int, long)):
+      raise TypeError('time must be a number; received %r' % time)
+    if namespace is None:
+      namespace = namespace_manager.get_namespace()
+    options = ('replace', time, namespace, deadline)
+    return self.memcache_set_batcher.add((key, value), options)
+
+  def memcache_cas(self, key, value, time=0, namespace=None, deadline=None):
+    if not isinstance(key, basestring):
+      raise TypeError('key must be a string; received %r' % key)
+    if not isinstance(time, (int, long)):
+      raise TypeError('time must be a number; received %r' % time)
+    if namespace is None:
+      namespace = namespace_manager.get_namespace()
+    return self.memcache_set_batcher.add((key, value),
+                                          ('cas', time, namespace, deadline))
+
+  def memcache_delete(self, key, seconds=0, namespace=None, deadline=None):
+    if not isinstance(key, basestring):
+      raise TypeError('key must be a string; received %r' % key)
+    if not isinstance(seconds, (int, long)):
+      raise TypeError('seconds must be a number; received %r' % seconds)
+    if namespace is None:
+      namespace = namespace_manager.get_namespace()
+    return self.memcache_del_batcher.add(key, (seconds, namespace, deadline))
+
+  def memcache_incr(self, key, delta=1, initial_value=None, namespace=None,
+                    deadline=None):
+    if not isinstance(key, basestring):
+      raise TypeError('key must be a string; received %r' % key)
+    if not isinstance(delta, (int, long)):
+      raise TypeError('delta must be a number; received %r' % delta)
+    if initial_value is not None and not isinstance(initial_value, (int, long)):
+      raise TypeError('initial_value must be a number or None; received %r' %
+                      initial_value)
+    if namespace is None:
```

Comment:
I'd consider moving the namespace out of the client.

Reply:
Yes, absolutely. Maybe as a follow-on change to this? :)

Reply:
SGTM